lib/rubocop/cop/sorbet/scope_helper_position.rb

```diff
@@ -0,0 +1,89 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # This cop ensures that helpers such as abstract!, interface! and final!
+      # are invoke before any method definitions or invocations
+      #
+      # @example
+      #
+      #   # bad
+      #   module Interface
+      #     extend T::Sig
+      #     extend T::Helpers
+      #
+      #     sig { returns(String) }
+      #     def foo; end
+      #
+      #     interface!
+      #   end
+      #
+      #   # good
+      #   module Interface
+      #     extend T::Sig
+      #     extend T::Helpers
+      #
+      #     interface!
+      #
+      #     sig { returns(String) }
+      #     def foo; end
+      #   end
+      class ScopeHelperPosition < RuboCop::Cop::Cop
+        include RangeHelp
+        HELPERS = %i(interface! abstract! final! sealed!).freeze
+        ALLOWED_METHODS = %i(extend include prepend requires_ancestor).freeze
+
+        def autocorrect(node)
+          lambda do |corrector|
+            corrector.remove(range_by_whole_lines(node.source_range, include_final_newline: true))
+
+            if processed_source[node.source_range.line - 2].empty? &&
+              (processed_source[node.source_range.line].empty? || processed_source[node.source_range.line] == "end")
+              corrector.remove(
+                range_by_whole_lines(source_range(processed_source.buffer, node.source_range.line - 1, 0))
+              )
+            end
+
+            before_node = node.parent.each_child_node.find do |child_node|
+              method_invocation = child_node.send_type? || child_node.block_type?
+              offense_invocation = method_invocation && !ALLOWED_METHODS.include?(child_node.method_name) &&
+                !HELPERS.include?(child_node.method_name)
+
+              child_node.def_type? || offense_invocation
+            end
```

Comment:
This seems to be common logic between the autocorrect and the check. Might be a good idea to extract this into a method.